schema/cam/cam_schema_2-1-0.json

```diff
@@ -170,12 +163,28 @@
                   }
                 },
                 "altitude": {
-                  "$comment": "if not provided, 'altitude' = 15 (unavailable)",
-                  "type": "integer",
-                  "description": "alt-000-01 (0), alt-000-02 (1), alt-000-05 (2), alt-000-10 (3), alt-000-20 (4), alt-000-50 (5), alt-001-00 (6), alt-002-00 (7), alt-005-00 (8), alt-010-00 (9), alt-020-00 (10), alt-050-00 (11), alt-100-00 (12), alt-200-00 (13), outOfRange (14), unavailable (15)",
-                  "default": 15,
-                  "minimum": 0,
-                  "maximum": 15
+                  "type": "object",
+                  "required": [
+                    "altitude_value",
+                    "altitude_confidence"
+                  ],
+                  "properties": {
+                    "altitude_value": {
+                      "type": "integer",
+                      "description": "Unit: 0.01 meter. referenceEllipsoidSurface(0), oneCentimeter(1), unavailable(800001)",
+                      "default": 800001,
+                      "minimum": -100000,
+                      "maximum": 800001
+                    },
+                    "altitude_confidence": {
+                      "$comment": "if not provided, 'altitude_confidence' = 15 (unavailable)",
```

Comment:
I wonder where this `$comment` is coming from; it certainly is neither from the TS nor from the EN documents. I also wonder whether it is helpful, and whether it is even correct.

Indeed, `altitude_confidence` is a mandatory property of the `altitude` object, so it can't be "not provided". As such, the comment seems to be contradicting the `required` section for the `altitude` object...

I know there were already such comments in the schema before this change, but no need to add more incorrect such comments...

(Note: this applies to the whole branch, of course, and I'll only comment on this one.)

@mathieu1fb Thoughts?

Reply:
> if a confidence field is mandatory in the spec, but an "unavailable" value exists for it, then we can consider this field optional in the JSON adaptation

I fully support this proposal! It makes so much sense! 👍

Reply:
> if a confidence field is mandatory in the spec, but an "unavailable" value exists for it, then we can consider this field optional in the JSON adaptation

Should we also apply this rule to other properties as well, not just confidence properties?

For example, in the `heading` object, both the `heading_value` and `heading_confidence` properties are required, and they both have an _unavailable_ value. In such a case, I'd say neither properties should be _required_, and thus that the whole `heading` object should also not be _required_ either.

Reply:
I would say that in this example, the `heading` object can then be optional, but if provided, then only its `value` field should be mandatory.